doc/grpctunnel_design.md

```diff
@@ -258,3 +329,41 @@ an ack to the tunnel client. The client will then check if it can handle the
 request, and if it can, it will create a new tunnel session to the server. The
 tunnel session is then forwarded to the client handler function on the client,
 and returned to the `NewSession` request on the server.
+
+### Target Registration
+
+During registration, a client will send a `Target` message to register its
+target (ID and type). The server will check if the target is already registered.
+It will return an error message if already existed, or return an accepted ack 
```

Comment:
an error message if the registration is not accepted (e.g. target already exists) or return an accepted acknowledgement message otherwise.